zerver/views/messages.py

```diff
@@ -1469,6 +1470,16 @@ def delete_message_backend(request: HttpRequest, user_profile: UserProfile,
     do_delete_message(user_profile, message)
     return json_success()
 
+@has_request_variables
+def delete_in_topic(request: HttpRequest, user_profile: UserProfile,
+                            stream_id: int=REQ('stream_id'),
+                            topic_name: str=REQ('topic_name')) -> None:
+    # TODO: add req param validators
+    print("LOG")
+    messages_in_topic = messages_for_topic(stream_id, topic_name)
+    for message in messages_in_topic:
+        do_delete_message(user_profile, message)
```

Comment:
You'll want to modify `do_delete_message` to support deleting a group of messages.  Unfortunately, the event design will require a loop for the events part, but you can at least use a bulk database query for the `move_messages_to_archive` part (which is most of the work).

Reply:
I would recommend renaming `do_delete_message` to `do_delete_messages` in a new commit adding at the start of this PR (before your changes), and adjust it to take an iterator of messages (even though the existing callers would only pass `[message]` for that argument).  And then this can call that.

Reply:
There is already a function named `do_delete_messages` [here](https://github.com/kunal-mohta/zulip/blob/master/zerver/lib/actions.py#L4108-L4111) :sweat_smile: 
Should I rename this?

Reply:
Yes, rename the existing one to `do_delete_messages_by_sender(user)`.  You can do that rename in a new first commit in the branch, following our commit guidelines.